Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -0,0 +1,194 @@
+
+    /// Creates a new compression configuration.
+    ///
+    /// - SeeAlso: ``defaults``.
+    public init(enabledAlgorithms: CompressionAlgorithmSet) {
+      self.enabledAlgorithms = enabledAlgorithms
+    }
+
+    /// Default values, compression is disabled.
+    public static var defaults: Self {
+      Self(enabledAlgorithms: .none)
+    }
+  }
+
+  public struct Keepalive: Sendable {
+    /// The amount of time to wait after reading data before sending a keepalive ping.
+    public var time: TimeAmount
+
+    /// The amount of time the server has to respond to a keepalive ping before the connection is closed.
+    public var timeout: TimeAmount
+
+    /// Whether the server allows the client to send keepalive pings when there are no calls in progress.
+    public var permitWithoutCalls: Bool
+
+    /// The minimum allowed interval the client is allowed to send keep-alive pings.
+    /// Pings more frequent than this interval count as 'strikes' and the connection is closed if there are
+    /// too many strikes.
+    public var minPingIntervalWithoutCalls: TimeAmount
+
+    /// Creates a new keepalive configuration.
+    public init(
+      time: TimeAmount,
+      timeout: TimeAmount,
+      permitWithoutCalls: Bool,
+      minPingIntervalWithoutCalls: TimeAmount
+    ) {
+      self.time = time
+      self.timeout = timeout
+      self.permitWithoutCalls = permitWithoutCalls
+      self.minPingIntervalWithoutCalls = minPingIntervalWithoutCalls
+    }
+  }
+
+  public struct Idle: Sendable {
+    /// The maximum amount of time a connection may be idle before it's closed.
+    public var maxTime: TimeAmount
+
+    /// Creates an idle configuration.
+    public init(maxTime: TimeAmount) {
+      self.maxTime = maxTime
+    }
+
+    /// Default values, a 30 minute max idle time.
+    public static var defaults: Self {
+      Self(maxTime: .seconds(30 * 60))
+    }
+  }
+
+  public struct Connection: Sendable {
+    /// The socket address to bind on.
+    public var socketAddress: NIOCore.SocketAddress
+
+    /// The maximum amount of time a connection may exist before being gracefully closed.
+    public var maxAge: TimeAmount?
+
+    /// The maximum amount of time that the connection has to close gracefully.
+    public var maxGraceTime: TimeAmount
+  }
+
+  public struct HTTP2: Sendable {
+
+    /// The maximum frame size to be used in an HTTP/2 connection.
+    public var maxFrameSize: Int
+
+    /// Whether TLS is being used or not in the connection.
+    public var useTLS: Bool
+
+    /// The target window size for this connection.
+    ///
+    /// - Note: This will also be set as the initial window size for the connection.
+    public var targetWindowSize: Int
+
+    /// The number of concurrent streams on the HTTP/2 connection.
+    public var maxConcurrentStreams: Int
+  }
+}
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension ChannelPipeline.SynchronousOperations {
```

Comment:
That file doesn't exist yet, it's added in your open PR. I can create the file too and we'll just have a conflict if you want? Otherwise I can move this when I tackle the TLS config in a follow-up.